### PR TITLE
fix(itinerary-body): positioning repair post-a11y work

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This PR corrects some clickable region bounds that broke with the recent a11y work. Also correct some general formatting issues.

### CODE REVIEW

File: packages/itinerary-body/src/styled.tsx [hunk 1]
```diff
@@ -571,15 +558,22 @@ export const StepDescriptionContainer = styled.div`
...
-  display: inline-block;
+  display: inline-flex;
   align-self: center;
   margin-top: 10px;
+
+  a {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+  }
```
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/styled.tsx:569]: What is this change for? It squeezes the map button in the screenshot and makes spacing inconsistent with the map button in the expanded steps.

image

File: packages/itinerary-body/src/styled.tsx [hunk 2]
```diff
@@ -286,6 +268,7 @@ export const LegClickableButton = styled(TransparentButton)`
   position: absolute;
   right: 0;
   top: 0;
+  width: 100%;
```
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/styled.tsx:271]: This doesn't do anything more, the width is already set by `left` and `right` under absolute positioning.

File: packages/itinerary-body/src/styled.tsx [hunk 3]
```diff
@@ -349,7 +332,7 @@ export const LegIconContainer = styled.span`
   img,
   svg {
     margin-right: 6px;
-    max-height: 24px;
+    height: 24px;
```
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/styled.tsx:335]: From the browser dev tools, this will not achieve the desired effect.
image of Chrome dev tools with grayed out image width and height props, and popup stating that the display:inline property prevents the size from taking effect.

File: packages/itinerary-body/src/styled.tsx [hunk 4]
```diff
@@ -483,6 +466,10 @@ export const PlaceSubheader = styled.div`
   font-weight: 300;
   padding-left: 4px;
   padding-top: 1px;
+
+  /* TODO: Why is this needed? */
```
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/styled.tsx:470]:
```suggestion
  /* Reduce vertical space and fix horizontal alignment of stop id and stop viewer link for transit stops. */
```

File: packages/itinerary-body/src/styled.tsx [hunk 5]
```diff
@@ -571,15 +558,22 @@ export const StepDescriptionContainer = styled.div`
...
+    align-items: center;
+    justify-content: center;
+  }
 `;
 
 export const StepsHeader = styled(TransparentButton)`
   color: #676767;
   font-size: 13px;
   font-style: normal;
+  width: 100%;
```
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/styled.tsx:576]: Revert this change, it makes for an odd layout (see screenshot)

image

File: packages/itinerary-body/src/styled.tsx (hunk 1 above)
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/styled.tsx:569]: I agree the new placement has some issues

File: packages/itinerary-body/src/styled.tsx [hunk 6]
```diff
@@ -676,6 +670,7 @@ export const TransitLegDetails = styled.div`
 
 export const TransitLegDetailsHeader = styled.div`
   color: #676767;
+  display: flex;
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/styled.tsx:673]: Removing this causes spacing issues on Safari

File: packages/itinerary-body/src/styled.tsx (hunk 2 above)
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/styled.tsx:271]: Removing this causes the button to only be a few pixels wide in Safari

File: packages/itinerary-body/src/styled.tsx (hunk 3 above)
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/styled.tsx:335]: Does this apply to `img` as well or just `svg`?

File: packages/itinerary-body/src/styled.tsx (hunk 5 above)
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/styled.tsx:576]: I'm not quite what this transparent button is or why we added it, but it's causing a lot of problems

File: packages/itinerary-body/src/styled.tsx (hunk 3 above)
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/styled.tsx:335]: Just SVG I think. Unless you want to make the SVG `display:inline-block`.

File: packages/itinerary-body/src/styled.tsx (hunk 5 above)
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/styled.tsx:576]: Clicking the button expands or collapses the turn-by-turn directions.

File: packages/itinerary-body/src/styled.tsx (hunk 1 above)
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/styled.tsx:569]: Ok, please add a comment that this accommodates Firefox and Safari.

Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/styled.tsx:569]: Also, for the mapillary button, could you change `padding-left` to 4 or 5px?

**Review by binh-dam-ibigroup — APPROVED** [review]
Please add the indicated comments for future reference.

File: packages/itinerary-body/src/styled.tsx (hunk 3 above)
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/styled.tsx:335]: Yeah I've found those warnings to be very buggy especially for more complex css

**Comment by miles-grant-ibigroup** [comment]
Thanks! All changes are addressed

File: packages/itinerary-body/src/styled.tsx [hunk 7]
```diff
@@ -483,6 +467,10 @@ export const PlaceSubheader = styled.div`
   font-weight: 300;
   padding-left: 4px;
   padding-top: 1px;
+
+  /* Reduce vertical space and fix horizontal alignment of stop id and stop viewer link for transit stops. */
+  margin-top: -14px;
+  margin-left: -4px;
```
Comment (amy-corson-ibigroup) [inline packages/itinerary-body/src/styled.tsx:473]: Instead of the negative `margin-left` here, can you just take off the `padding-left` above?

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks good! Thanks for doing this!

File: packages/itinerary-body/src/styled.tsx (hunk 7 above)
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/styled.tsx:473]: That's a really good catch thank you!
